src/components/Tag/index.tsx

```diff
@@ -1,140 +1,147 @@
+  white-space: nowrap;
+  border-radius: ${({ theme }) => theme.space['0.5']};
+  padding: 0 ${({ theme }) => theme.space['1']};
+  gap: ${({ theme }) => theme.space['1']};
+  width: fit-content;
+  height: 24px;
+  ${({ variant }) => variant}
 `
 
-const StyledText = styled('span')`
-  ${({ 'aria-disabled': disabled }) => disabled && disabledStyles}
-  color: ${({ theme }) => theme.colors.neutral.text};
-  font-size: 14px;
-  align-self: center;
-  max-width: 350px;
-  overflow: hidden;
-  white-space: pre;
-  text-overflow: ellipsis;
+const StyledTypography = styled(Typography)`
+  font-size: 12px;
+  font-weight: 500;
+  color: inherit;
 `
 
 const StyledTouchable = styled(Touchable, {
   shouldForwardProp: props => !['variant'].includes(props.toString()),
-})<{ variant: TagVariant }>`
-  margin-left: 16px;
+})<{ variant: Color }>`
   display: flex;
   align-items: center;
-  justify-content: center;
+  svg {
+    fill: ${({ theme }) => theme.colors.neutral.textWeak};
+  }
   &:hover,
   &:focus {
-    background-color: ${({ theme }) => theme.colors.primary.backgroundHover};
     svg {
-      fill: ${({ theme }) => theme.colors.primary.text};
+      fill: ${({ theme }) => theme.colors.neutral.textHover};
     }
   }
-  ${({ variant, theme }) =>
-    variant === 'bordered' &&
-    `
-  border-radius: 4px;
-  border: 1px solid ${theme.colors.neutral.borderWeak};
-  padding: 4px;
-  width: 32px;
-  height: 32px;
-`}
 `
+const generateStyles = ({
+  theme,
+}: {
+  theme: Theme
+}): Record<string, string> => {
+  const text = 'text'
+  const background = 'background'
+
+  return {
+    ...SENTIMENTS.reduce(
+      (reducer, sentiment) => ({
+        ...reducer,
+        [sentiment]: `
+      color: ${theme.colors[sentiment][text]};
+      background: ${theme.colors[sentiment][background]}
+    `,
+      }),
+      {},
+    ),
+    disabled: `
+      color: ${theme.colors.neutral.textWeak};
+      background: ${theme.colors.neutral.backgroundStrong};
+    `,
+    neutral: `
+      color: ${theme.colors.neutral.text};
+      background: ${theme.colors.neutral[background]};
+      border: solid 1px ${theme.colors.neutral.border};
+    `,
+  }
+}
 
 type TagProps = {
-  children: ReactNode
-  disabled?: boolean
   isLoading?: boolean
   onClose?: MouseEventHandler<HTMLButtonElement>
   textStyle?: JSX.IntrinsicAttributes['css']
-  variant?: TagVariant
-} & BoxProps
+  variant?: Color
+  /**
+   * Defines icon to display on left side of badge. **Only available on medium and large sizes**.
+   */
+  disabled?: boolean
+  icon?: IconName
+  className?: string
+  children: ReactNode
+}
 
 const Tag = ({
   children,
   isLoading = false,
   onClose,
   textStyle,
+  icon,
   disabled = false,
-  variant = 'base',
-  ...props
-}: TagProps) => (
-  <StyledContainer {...props} disabled={disabled} variant={variant}>
-    <StyledText aria-disabled={disabled} css={textStyle}>
-      {children}
-    </StyledText>
+  variant = 'neutral',
+  className,
```

Comment:
no issues to dropping xsytled props ?

Reply:
I tested it with yalc and there is some `tag` in shire effected, so I created an MR to migrate